src/Symfony/Component/Uid/Command/GenerateUlidCommand.php

```diff
@@ -0,0 +1,119 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Uid\Command;
+
+use Symfony\Component\Console\Command\Command;
+use Symfony\Component\Console\Input\InputInterface;
+use Symfony\Component\Console\Input\InputOption;
+use Symfony\Component\Console\Output\ConsoleOutputInterface;
+use Symfony\Component\Console\Output\OutputInterface;
+use Symfony\Component\Console\Style\SymfonyStyle;
+use Symfony\Component\Uid\Factory\UlidFactory;
+use Symfony\Component\Uid\Ulid;
+
+class GenerateUlidCommand extends Command
+{
+    protected static $defaultName = 'ulid:generate';
+    protected static $defaultDescription = 'Generates a ULID';
+
+    private $factory;
+
+    public function __construct(UlidFactory $factory = null)
+    {
+        $this->factory = $factory ?? new UlidFactory();
+
+        parent::__construct();
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    protected function configure(): void
+    {
+        $this
+            ->setDefinition([
+                new InputOption('timestamp', null, InputOption::VALUE_REQUIRED, 'The ULID timestamp: a parsable date/time string by the \DateTimeImmutable constructor. It must be greater than or equals to the UNIX epoch (1970-01-01 00:00:00)'),
+                new InputOption('count', 'c', InputOption::VALUE_REQUIRED, 'The number of ULID to generate', 1),
+                new InputOption('format', 'f', InputOption::VALUE_REQUIRED, 'The ULID output format: canonical, base_58 or rfc_4122', 'canonical'),
+            ])
+            ->setDescription(self::$defaultDescription)
+            ->setHelp(<<<'EOF'
+The <info>%command.name%</info> command generates a ULID.
+
+    <info>php %command.full_name%</info>
+
+To specify the timestamp:
+
+    <info>php %command.full_name% --timestamp="2021-02-16 14:09:08"</info>
+
+To generate several ULIDs:
+
+    <info>php %command.full_name% --count=10</info>
+
+To output a specific format:
+
+    <info>php %command.full_name% --format=rfc_4122</info>
+EOF
+            )
+        ;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    protected function execute(InputInterface $input, OutputInterface $output)
+    {
+        $io = new SymfonyStyle($input, $output instanceof ConsoleOutputInterface ? $output->getErrorOutput() : $output);
+
+        if (null !== $time = $input->getOption('timestamp')) {
+            try {
+                $time = new \DateTimeImmutable($time);
+            } catch (\Exception $e) {
+                $io->error(sprintf('Invalid timestamp "%s". It cannot be parsed.', $time));
```

Comment:
maybe reuse $e->getMessage(), after removing the `DateTimeImmutable::__construct(): ` prefix?